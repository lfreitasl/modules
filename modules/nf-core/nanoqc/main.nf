process NANOQC {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/nanoqc:0.10.0--pyhdfd78af_0':
        'biocontainers/nanoqc:0.10.0--pyhdfd78af_0' }"

    input:
    tuple val(meta), path(fastq)

    output:
    tuple val(meta), path("*.html"), emit: report
    tuple val(meta), path("*.log") , emit: log
    path "versions.yml"            , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args   = task.ext.args   ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    nanoQC \\
        $fastq \\
        $args
    
    mv *.html ${prefix}.html
    mv *.log ${prefix}.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        nanoqc: \$(nanoQC --version |& sed '1!d ; s/NanoQC //')
    END_VERSIONS
    """

    stub:
    def args   = task.ext.args   ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    
    """
    touch ${prefix}.html
    touch ${prefix}.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        nanoqc: \$(nanoQC --version |& sed '1!d ; s/NanoQC //')
    END_VERSIONS
    """
}
