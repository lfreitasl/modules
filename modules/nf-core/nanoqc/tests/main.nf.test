
nextflow_process {

    name "Test Process NANOQC"
    script "../main.nf"
    process "NANOQC"

    tag "modules"
    tag "modules_nfcore"
    tag "nanoqc"

    test("sarscov2 - fastq") {

        
        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/fastq/test.fastq.gz', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
                
            )
        }

    }


    test("sarscov2 - fastq - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/nanopore/fastq/test.fastq.gz', checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out.versions).match() }
            )
        }

    }

}
