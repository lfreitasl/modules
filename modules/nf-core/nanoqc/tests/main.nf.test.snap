{
    "sarscov2 - fastq - stub": {
        "content": [
            [
                "versions.yml:md5,a3dafd8e8710de433af43795c43146bb"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-11T19:52:21.921878306"
    },
    "sarscov2 - fastq": {
        "content": [
            [
                "versions.yml:md5,a3dafd8e8710de433af43795c43146bb"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.0"
        },
        "timestamp": "2024-11-11T19:52:12.180643962"
    }
}